spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/PropertiesMeterFilter.java

```diff
@@ -79,11 +82,13 @@ public Id map(Id id) {
 	public DistributionStatisticConfig configure(Meter.Id id,
 			DistributionStatisticConfig config) {
 		Distribution distribution = this.properties.getDistribution();
-		return DistributionStatisticConfig.builder()
-				.percentilesHistogram(
-						lookup(distribution.getPercentilesHistogram(), id, null))
-				.percentiles(lookup(distribution.getPercentiles(), id, null))
-				.sla(convertSla(id.getType(), lookup(distribution.getSla(), id, null)))
+		return DistributionStatisticConfig.builder().percentilesHistogram(lookup(
+				distribution.getPercentilesHistogram(), id, null,
+				(all) -> distribution.getPercentilesHistogram().getOrDefault(all, null)))
```

Comment:
I'm not too keen on the repetition of `distribution.getPercentilesHistogram()`.

I think it would be better to have a `lookup` method and another method that's something like `lookupWithFallbackToAll`. Both could delegate to another method to do the actual lookup and could then either just apply the default value, or could fall back to looking for `all` and only then applying the default value if needed.

That would change this code to look something like this:

```java
return DistributionStatisticConfig.builder()
		.percentilesHistogram(lookupWithFallbackToAll(
				distribution.getPercentilesHistogram(), id, null))
		.percentiles(
				lookupWithFallbackToAll(distribution.getPercentiles(), id, null))
		.sla(convertSla(id.getType(), lookup(distribution.getSla(), id, null)))
		.build().merge(config);
```